HTTP Server.core/Reply.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Text;
+
+namespace HTTPServer.core
+{
+    public static class Reply
+    {
+        public static byte[] HandleReply(byte[] request, IDirectoryContents directoryContents)
+        {
+            var requestMessage = Encoding.UTF8.GetString(request).Substring(0, request.Length);
+            var uri = "";
+            var method = "";
+            var version = "";
+            try
+            {
+                string[] requestLine = requestMessage.Split(' ', ' ');
+                method = requestLine[0];
+                uri = requestLine[1];
+                version = requestLine[2];
+            }
+            catch
+            {
+                return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 400 Bad Request\r\n", directoryContents));
+            }
+            if (version.Equals("HTTP/1.1\r\n") && IsRequestMethod(method))
+            {
+                if (uri.Equals("/"))
+                    return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 200 OK\r\n", directoryContents));
+                return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 404 Not Found\r\n", directoryContents));
+            }
+            return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 400 Bad Request\r\n", directoryContents));
+        }
+        
+        private static bool IsRequestMethod(string str)
+        {
+            return str.Equals("GET");
+        }
+
+        public static string CreateMessage(string httpStartLine, IDirectoryContents directoryContents)
+        {
+            if (directoryContents.DirectoryPath.Equals(""))
+                return httpStartLine;
+            string[] files = directoryContents.GetFiles();
+            string[] directories = directoryContents.GetDirectories();
+            var wholeMessage = httpStartLine;
+            var bodyMessage = "<html>\r\n" +
```

Comment:
you don't need the newlines in the html. this would also simplify your test data a little